components/Templates/includes/functions-view_template.php

```diff
@@ -297,30 +297,117 @@ function frontier_do_subtemplate( $atts, $content ) {
 				}
 
 				$out .= pods_shortcode( array(
-					'name'  => $pod->fields[ $atts[ 'field' ] ][ 'pick_val' ],
+					'name'  => $field[ 'pick_val' ],
 					'slug'  => $target_id,
 					'index' => $key
 				), $template );
 
 			}
-		} elseif ( 'file' == $pod->fields[ $atts[ 'field' ] ][ 'type' ] && 'attachment' == $pod->fields[ $atts[ 'field' ] ][ 'options' ][ 'file_uploader' ] && 'multi' == $pod->fields[ $atts[ 'field' ] ][ 'options' ][ 'file_format_type' ] ) {
+		} elseif ( 'file' == $field[ 'type' ] && 'attachment' == $field[ 'options' ][ 'file_uploader' ] && 'multi' == $field[ 'options' ][ 'file_format_type' ] ) {
 			$template = frontier_decode_template( $content, $atts );
 			foreach ( $entries as $key => $entry ) {
 				$content = str_replace( '{_index}', $key, $template );
 				$content = str_replace( '{@_img', '{@image_attachment.' . $entry[ 'ID' ], $content );
 				$content = str_replace( '{@_src', '{@image_attachment_url.' . $entry[ 'ID' ], $content );
-				$content = str_replace( '{@' . $atts[ 'field' ] . '}', '{@image_attachment.' . $entry[ 'ID' ] . '}', $content );
+				$content = str_replace( '{@' . $field_name . '}', '{@image_attachment.' . $entry[ 'ID' ] . '}', $content );
 
 				$out .= pods_do_shortcode( $pod->do_magic_tags( $content ), frontier_get_shortcodes() );
 			}
 
+		} else {
+			// Relationship to something other than a Pod (ie: user)
+			foreach ( $entries as $key => $entry ) {
+				$template = frontier_decode_template( $content, $atts );
+				$template = str_replace( '{_index}', $key, $template );
+				if ( !is_array( $entry ) ) {
+					$entry = array( '_key' => $key, '_value' => $entry );
+				}
+				$out .= pods_do_shortcode( frontier_pseudo_magic_tags( $template, $entry, $pod ), frontier_get_shortcodes() );
+			}
 		}
 	}
 
 	return pods_do_shortcode( $out, frontier_get_shortcodes() );
 }
 
 /**
+ *
```

Comment:
Why don't we add something like this into a more reusable function outside of the frontier namespace. We might even find a way to abstract some of this logic so both normal magic tags processing and this one aren't too different?

Reply:
If we can do it reasonably safely in time for 2.7.  If not, I'd vote for saving that for a big rework of templates.

Reply:
I thought about this but then decided against it because I didn't want to encourage anyone to go down the dead end of the existing magic tag system.

Reply:
Fair enough, magic tags aren't super amazing anyway